pages/en/developer/create-subgraph-hosted.mdx

```diff
@@ -210,7 +210,9 @@ Entity fields can be defined as required or optional. Required fields are indica
 Null value resolved for non-null field 'name'
 ```
 
-Each entity must have an `id` field, which is of type `ID!` (string). The `id` field serves as the primary key, and needs to be unique among all entities of the same type.
+Each entity must have an `id` field, which must be of type `Bytes!` or `String!`. It is generally recommended to use `Bytes!`, unless the `id` contains human-readable text, since entities with `Bytes!` id's will be faster to write and query as those with a `String!` `id`. The `id` field serves as the primary key, and needs to be unique among all entities of the same type. For historical reasons, the type `ID!` is also accepted and is a synonym for `String!`.
```

Comment:
I think this is a typo:
```
`String!` `id`. The `id`
```
Should be:
```
`String!`. The `id`
```

Reply:
No, I meant to say 'string id' here. All the backquotes make it a little hard to see, but if I read the sentence out loud, it makes sense to me as written.